src/components/atoms/Accordion/Accordion.tsx

```diff
@@ -1,96 +1,144 @@
-import React, { useState, useEffect, useRef } from 'react';
+import React, { useState, useEffect, useRef, useCallback } from 'react';
 
-import { KeyboardArrowDown, KeyboardArrowUp } from '@groww-tech/icon-store/mi';
+import { KeyboardArrowDown } from '@groww-tech/icon-store/mi';
 
 import AnimateHeight from '../AnimateHeight';
 
 import './accordion.css';
 
 
 const Accordion = (props: Props) => {
+  const {
+    onMountOpen,
+    onToggleCallback,
+    title,
+    children,
+    parentClass,
+    headerClass,
+    iconClass,
+    titleClass,
+    showRightIcon,
+    useAnimateHeight,
+    maxHeight
+  } = props;
+
+  const [isOpen, toggleAccordion] = useState(onMountOpen);
 
+  const childRef = useRef<HTMLDivElement>(null);
   const isFirstRender = useRef(true);
-  const [ isOpen, setAccordionToggle ] = useState(props.onMountOpen);
+
 
   useEffect(() => {
     if (isFirstRender.current) {
       isFirstRender.current = false;
       return;
     }
 
-    props.onToggleCallback(isOpen);
-  }, [ isOpen ]);
+    if (onToggleCallback) {
+      onToggleCallback(isOpen);
+    }
+  }, [isOpen]);
 
 
-  const toggleState = () => {
-    setAccordionToggle(!isOpen);
-  };
+  const toggleState = useCallback(() => {
+    toggleAccordion(isOpen => !isOpen);
+  }, [isOpen]);
```

Comment:
We don't need isOpen dependency here.
isOpen inside useCallback is scoped variable. we can rename it.